docs/contributing.md

```diff
@@ -2,9 +2,58 @@
 
 Thank you for your interest!
 
-## Running tests
-* `python -m pip install pytest`
-* `python -m pytest tests`
+## Setting up the dev environment
+
+It is ideal to create a virtual environment whenever you want to work on
+a python project. **Note:** This is not a necessity to work on sanic.
+
+Create a a new virtual environment if you dont have one, using:
+
+`python3 -m venv <any name>`
+
+Enable it using:
+
+`. <any name>/bin/activate`
+
+Get the source of sanic:
+
+`git clone https://github.com/channelcat/sanic`
+
+`cd sanic`
+
+Install sanic :
+
+`python setup.py install`
```

Comment:
This is actually incorrect, if you want to do development on any python project you should do `python setup.py develop` or `pip install -e .`